File: spring-boot/src/main/java/com/example/todo/dto/CreatePostRequest.java
package com.example.todo.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

@Schema(description = "Request to create a new post")
public class CreatePostRequest {

    @NotBlank(message = "Discussion ID is required")
    @Schema(description = "ID of the discussion this post belongs to", example = "507f1f77bcf86cd799439011", required = true)
    private String discussionId;

    @NotBlank(message = "Course ID is required")
    @Schema(description = "ID of the course this post belongs to", example = "507f1f77bcf86cd799439011", required = true)
    private String courseId;

    @NotBlank(message = "Post content is required")
    @Size(min = 1, max = 5000, message = "Post content must be between 1 and 5000 characters")
    @Schema(description = "Content of the post", example = "This is a great question!", required = true)
    private String content;

    // Getters and Setters
    public String getDiscussionId() {
        return discussionId;
    }

    public void setDiscussionId(String discussionId) {
        this.discussionId = discussionId;
    }

    public String getCourseId() {
        return courseId;
    }

    public void setCourseId(String courseId) {
        this.courseId = courseId;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }
}

File: aspnet-core-mvc/DTOs/CreatePostRequest.cs
using System.ComponentModel.DataAnnotations;

namespace CourseManagementAPI.DTOs;

public class CreatePostRequest
{
    [Required(ErrorMessage = "Discussion ID is required")]
    public string DiscussionId { get; set; } = string.Empty;

    [Required(ErrorMessage = "Course ID is required")]
    public string CourseId { get; set; } = string.Empty;

    [Required(ErrorMessage = "Post content is required")]
    [MinLength(1, ErrorMessage = "Content must be at least 1 character")]
    [MaxLength(5000, ErrorMessage = "Content must not exceed 5000 characters")]
    public string Content { get; set; } = string.Empty;
}

